x-pack/plugins/actions/server/builtin_action_types/lib/send_email_graph_api.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import axios, { AxiosResponse } from 'axios';
+import { Logger } from '../../../../../../src/core/server';
+import { request } from './axios_utils';
+import { ActionsConfigurationUtilities } from '../../actions_config';
+import { SendEmailOptions } from './send_email';
+
+interface SendEmailGraphApiOptions {
+  options: SendEmailOptions;
+  headers: Record<string, string>;
+  messageHTML: string;
+  graphApiUrl?: string;
+}
+
+const MICROSOFT_GRAPH_API_HOST = 'https://graph.microsoft.com/v1.0';
+
+export async function sendEmailGraphApi(
+  sendEmailOptions: SendEmailGraphApiOptions,
+  logger: Logger,
+  configurationUtilities: ActionsConfigurationUtilities
+): Promise<AxiosResponse> {
+  const { options, headers, messageHTML, graphApiUrl } = sendEmailOptions;
+
+  const axiosInstance = axios.create();
+
+  // POST /users/{id | userPrincipalName}/sendMail
+  const res = await request({
+    axios: axiosInstance,
+    url: `${graphApiUrl ?? MICROSOFT_GRAPH_API_HOST}/users/${options.routing.from}/sendMail`,
+    method: 'post',
+    logger,
+    data: getMessage(options, messageHTML),
+    headers,
+    configurationUtilities,
+    validateStatus: () => true,
+  });
+  if (res.status === 202) {
```

Comment:
Are these status codes in the Microsoft documentation somewhere that we can link to? For future developers :). I don't see 202s very often

Reply:
Yeah, I will add the comment 👍 https://docs.microsoft.com/en-us/graph/api/user-sendmail?view=graph-rest-1.0&tabs=http#response-4